README.md

```diff
@@ -224,28 +224,118 @@ particlePositions(:,:,2) = particlePositionsY;
+
+*Input* - `particlesMat` - The current position of the particles, `R` - Particle radius, `D` - Diffusion coefficient from Stokes-Einstein relation [*You can see we assume our particles are spherical*].
+*Output* - `DMat` - A diffusion matrice relating every axis of every particle.
+*Inside* - Basically converting the math from the articles to code. You can check out equation (8) in [[3](#references)].
+
+---
+
+### Self defined functions
+All these functions exist to add extra options when running the simulation.
+
+#### feedbackCheckFunc
+This function checks conditions set by the user, and if it fits the set conditions it runs the `feedbackFunc`.
+
+*Input* - `simConfig`,`simStepdata`,`additionalData` - All our relevant variables.
+*Output* - `Boolean`
+
+Example -  
+```matlab
+function f = checkIfMoveWall(cfg, currStepData, addedData)
+    f = False;
+    rightmostPosition = max(currStepData.particlePositions(:,1))+cfg.R(1);
+    addedData.closestParticlePositions = rightmostPosition;
+    if currStepData.wallPositionsX(2) - rightmostPosition > 2*cfg.R(1)
+	    f = True;
+	end
+```
+Here our `feedbackCheckFunc` is `checkIfMoveWall`. The function checks if the distance between the rightmost particle and the wall on that side is larger than 3 particle radii. If it is we just go into running the `feedbackFunc`. We also save the position of that particle into `addedData.closestParticlePositions`.
+
+
+####  feedbackFunc
+This function runs after passing the conditions set in `feedbackCheckFunc`
+
+*Input* - `simConfig`,`simStepdata`,`additionalData` - All our relevant variables.
+*Output* - Your choice.
+
+Example -  
+```matlab
+function moveWall(cfg, currStepData, addedData)
+	currStepData.wallPositionsX(2) = addedData.closestParticlePositions;
+```
+After confirming that the distance of the particle to the wall was what we wanted we move the wall to 1 radius away from the center of the rightmost particle. We use our `addedData` to transfer the information from the `feedbackCheckFunc` into the `feedbackFunc`
+
+####  printFunc
+Every time we reach a sampling period we will have a chance to use our defined `printFunc` if we set `DisplayLive` to True in the configuration file.
+
+*Input* - `simConfig`,`simStepdata`,`additionalData` - All our relevant variables.
+*Output* - Your choice.
+
+```matlab
+function printTrapPosition(cfg, currStepData, addedData)
+    cla
+    hold all
+    xlim(cfg.xlimits)
+	ylim(cfg.ylimits)
+    for i = 1:size(cfg.initTrapPositions,1)
+		plot(currStepData.trapPositions(1,i),currStepData.trapPositions(2,i),'o'); 
+    end
+    pause(1); 
+```
+Using the limits we set in the configuration file we plot all of the traps current places.
 
 ## Basic examples
 
-Please read [Examples.md](https://gist.github.com/PurpleBooth/b24679402957c63ec426) for some basic usage examples
+Please read [Examples.md](http) for some basic usage examples
```

Comment:
here please change the like to the relevant one.

Reply:
Not sure what you mean, since as far as I know there isn't a relevant one yet at the moment